app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -10,15 +10,41 @@ import android.support.v4.app.FragmentManager
 import android.text.TextUtils
 import mozilla.components.browser.session.Session
 import org.mozilla.tv.firefox.webrender.WebRenderFragment
-import org.mozilla.tv.firefox.ext.webRenderComponents
+import org.mozilla.tv.firefox.navigationoverlay.NavigationOverlayFragment
 import org.mozilla.tv.firefox.pocket.PocketVideoFragment
 import org.mozilla.tv.firefox.settings.SettingsFragment
 import org.mozilla.tv.firefox.telemetry.TelemetryIntegration
 import org.mozilla.tv.firefox.telemetry.UrlTextInputLocation
+import org.mozilla.tv.firefox.utils.URLs
 import org.mozilla.tv.firefox.utils.UrlUtils
 import org.mozilla.tv.firefox.widget.InlineAutocompleteEditText
 
-class ScreenController {
+class ScreenController(private val stateMachine: ScreenControllerStateMachine) {
+
+    // TODO cache overlay fragment so that custom pinned sites don't fade in every time the menu is opened
+
+    /**
+     * TODO
+     *
+     * not adding to backstack is intentional
+     */
+    fun setUpFragmentsForNewSession(fragmentManager: FragmentManager, session: Session) {
+        val renderFragment = WebRenderFragment.createForSession(session)
+        val pocketFragment = PocketVideoFragment()
+        val settingsFragment = SettingsFragment()
+        fragmentManager
+            .beginTransaction()
+            .add(R.id.container_settings, settingsFragment, SettingsFragment.FRAGMENT_TAG)
+            .add(R.id.container_pocket, pocketFragment, PocketVideoFragment.FRAGMENT_TAG)
+            .add(R.id.container_web_render, renderFragment, WebRenderFragment.FRAGMENT_TAG)
+            // We add NavigationOverlayFragment last so that it takes focus
+            .add(R.id.container_navigation_overlay, NavigationOverlayFragment(), NavigationOverlayFragment.FRAGMENT_TAG)
+            .hide(renderFragment) // TODO note that this will need to be changed in order to display WebRenderFragment under a split overlay
```

Comment:
Fair enough. fwiw, `Overlay` is the most important one because it's keeping large images in memory.

For context on what I ran into on Show, see https://github.com/mozilla-mobile/firefox-echo-show/issues/160

Reply:
Linked to this comment in #1310. But hopefully, removing the other fragments once we have channels will mitigate perf problems as seen on FFES.